sql/connect/common/src/main/scala/org/apache/spark/sql/connect/SparkSession.scala

```diff
@@ -789,6 +789,7 @@ object SparkSession extends SparkSessionCompanion with Logging {
         Option(System.getenv("SPARK_LOG_DIR"))
           .orElse(Option(System.getenv("SPARK_HOME")).map(p => Paths.get(p, "logs").toString))
           .foreach { p =>
+            Files.createDirectories(Paths.get(p))
```

Comment:
cc @HyukjinKwon 
It seems that manually creating the service directory here can resolve the issue, and even if the parent directory already exists, this action will not cause any actual negative impact.

Reply:
Or could we call `waitUntilFileExists(logDir)` before `waitUntilFileExists(logFile)`? Is it necessary to do so?